plugin.video.ampya/default.py

```diff
@@ -0,0 +1,341 @@
+
+def ListeChannels(url):
+  content=getUrl(url)
+  struktur = json.loads(content)
+  for element in struktur:   
+    title=element["asset"]["title"]
+    kuenstler_name=element["asset"]["display_artist_title"]
+    kuenstler_id=element["asset"]["artist_id"]
+    video_file_id=element["asset"]["video_file_id"]
+    Token=element["asset"]["token"]
+    duration=element["asset"]["duration"]
+    iconimage=getbild(video_file_id)
+    id=element["asset"]["id"]
+
+    
+    
+    debug("VIDEOID: "+str(video_file_id))
+    debug("iconimage :"+iconimage)
+    
+    
+    addLink(title +"("+kuenstler_name+")", Token, "playvideo", iconimage, duration=duration, desc="",artist_id=kuenstler_id,liedid=id)
+    debug("--------------")
+  xbmcplugin.endOfDirectory(addon_handle,succeeded=True,updateListing=False,cacheToDisc=True)   
+
+def getbild(video_file_id):
+    longid="%07d"%video_file_id
+    einsbisdrei=str(longid)[0:5]
+    iconimage="http://files.putpat.tv/artwork/posterframes/"+ einsbisdrei +"/" + longid +"/v"+longid+"_posterframe_putpat_large.jpg"
+    return iconimage
+    
+def ListeVideos(url):
+  content=getUrl(url)
+  struktur = json.loads(content)
+  for element in struktur:   
+    title=element["asset"]["title"]
+    kuenstler_name=element["asset"]["display_artist_title"]
+    kuenstler_id=element["asset"]["artist_id"]
+    video_file_id=element["asset"]["video_file_id"]
+    Token=element["asset"]["token"]
+    #duration=element["asset"]["duration"]
+    iconimage=getbild(video_file_id)
+    id=element["asset"]["id"]
+    
+    debug("VIDEOID: "+str(video_file_id))
+    debug("iconimage :"+iconimage)
+    
+    
+    addLink(title +"("+kuenstler_name+")", Token, "playvideo", iconimage, desc="",artist_id=kuenstler_id,liedid=id)
+    debug("--------------")
+  xbmcplugin.endOfDirectory(addon_handle,succeeded=True,updateListing=False,cacheToDisc=True)   
+
+def Playvideo(token):
+  file="https://www.putpat.tv/ws.json?method=Asset.getClipForToken&token="+token+"&streaming_method=http&client=android_phone&player_id=2"
+  content=getUrl(file)
+  struktur = json.loads(content)
+  file2=struktur[0]["clip"]["tokens"]["medium"]
+  debug("File2 : "+file2)
+  listitem = xbmcgui.ListItem(path=file2)  
+  xbmcplugin.setResolvedUrl(addon_handle,True, listitem)  
+
+def Listekanaele(url):
+  content=getUrl(url)
+  struktur = json.loads(content)
+  for name in struktur:
+   debug("....")
+   debug(name["channel"])
+   debug("....")
+   id=name["channel"]["id"]
+   title=name["channel"]["title"]
+   addLink(title , str(id), "playKanal", "http://files.putpat.tv/artwork/channelgraphics/"+str(id)+"/channellogo_150.png")
+  xbmcplugin.endOfDirectory(addon_handle,succeeded=True,updateListing=False,cacheToDisc=True) 
+  
+def monitor_playlist(cliplist,playlist,url,id)   :
+    counter=0
+    counter2=0
+    while counter2<5:
+        while xbmc.Player().isPlaying():
+            counter2=0
```

Comment:
Would monitor.waitForAbort(xx) run even if the Playlist Comes to an end?

so it must be somthing like "not monitor.abortRequested()" and xbmc.Player().isPlaying()

Right?